src/rebar_file_utils.erl

```diff
@@ -273,6 +275,22 @@ canonical_path([_|Acc], [".."|Rest])  -> canonical_path(Acc, Rest);
 canonical_path([], [".."|Rest])       -> canonical_path([], Rest);
 canonical_path(Acc, [Component|Rest]) -> canonical_path([Component|Acc], Rest).
 
+%% returns canonical target of path if path is a link, otherwise returns path
+-spec resolve_link(string()) -> string().
+
+resolve_link(Path) ->
+    case file:read_link(Path) of
+        {ok, Target} ->
+            canonical_path(filename:absname(Target, filename:dirname(Path)));
+        {error, _} -> Path
+    end.
```

Comment:
One small problem with this is that we usually fall back to a copy of the files when symlinking is not available on a system (i.e. an older windows). In this case, even submitting the `relative` option will mean you won't get it, since the symlink is accidental.

Another thing to consider is that if I'm developing in the `test` profile, a dependency from the `default` profile will be symlinked into the `test` profile, meaning the substitution may be wrong. It would be good to explicitly test how that goes since we have the possibility of:

- a symlink to a file in _build (as a dependency in a profile)
- a symlink to a file in src/ or apps/*/src (as an app in default)
- a symlink to a symlink in _build to a file in src/ or apps/*/src  (as an app in test -- though I'm not sure if that actually happens or not)